### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by franciscop** [franciscop/statux#2]
Suggested improvements:

- Document the persisting store. This is still alpha-ish (not documented on purpose) but from my testings, it's pretty good. The rest of the suggestions are to work around the shortcomings.
- Allow for null reads? Right now if you do `useSelector('users.35.id')` and 35 doesn't exists, it crashes. While the read would theoretically have no problem with returning null, the problem is on the writing, since if we try to write that id is on an object or array. However as [can be seen in this fiddle](https://jsfiddle.net/franciscop/2fd9347s/), arrays with custom properties do NOT survive stringification (however an object with a numeric key does survive). `users.0.id` could be `users:[]` or `users:{}`. Which brings us to:
- Hinting the type somehow? Default to objects since those also accept numbers, then for arrays allow to indicate them like one of these: `useStore('users:array.35.id');`, `useStore('[users].35.id')`, `useStore('users[].35.id')`, `useStore('users[35].id')`. I particularly like the last one since _it follows the JS syntax_, and it _could be done (probably) in a non-breaking way_. So you'd have the `.` for objects, and `[]` for arrays.
- To consider: placeholders to prevent potential injection. I expect people to do ``useSelector(`users.${id}.name`)`` or similar, but what if `id === '35.friends.0'`? You'd be getting the best friend's name instead. Since we have the simplest/dumbest query matching (sub->sub->sub etc) this might not be useful/make sense, e.g. `;` is not a comment and you cannot do `..` or similar. The solution is ALSO pretty simple with placeholders like `useSelector('users.{id}.name', { id })` or `useSelector('users.$id.name', { id })` (don't like this last one too much since $ means persistency). Main question is whether we even need to worry about this topic at this level or not.
- To consider: allow to use [`polystore`](https://polystore.dev/) as the data backing. This *might be impossible* since we'd need to wait on first load, and cascading `await` is very tricky+dangerous. Even with deferred saving, first load is still an issue.

Anti-features:

- We *cannot* add `useStore('key', { persist: true })` since this suffers from inconsistent-usage-order issues (what if an `useStore()` in a different place does NOT have the persistence? so this HAS to be in the highest level). Instead, suggested to either do it at the `<Store $key={...}>` level (as is now), or use a localStorage solution (useLocalState()-ish) from a different package. The above suggestions also help with this a bit (easier to do a general prefix).

**Comment by franciscop** [comment]
Don't want to create a custom language now, I've used the base project without these changes and it's worked great so far
